### PR TITLE
Enable doclint for javadocs.

### LINKED ISSUE #3435: Make Javadoc checks stricter
From #3419:

> Make our javadoc checks stricter. I think the best approach is to increase the strictness in client.pom.xml, and then lessen the strictness in the track one modules. Refer to https://blog.joda.org/2014/02/turning-off-doclint-in-jdk-8-javadoc.html and consider `Xdoclint:all`

There is an additional comment in https://github.com/Azure/azure-sdk-for-java/issues/3400#issuecomment-484298399 for an opt-in model for Track 1 SDKs and require Track 2 SDKs to adhere to this.

### PR DESCRIPTION
**Pull request by conniey**
* Adds doclint:all to javadoc plugins.
* By default, bad javadoc will fail the build. Exceptions were added for track 1 client libraries: storage, batch, keyvault.
  * Event-hubs was an easy fix, so it fails if there is bad javadoc.

Fixes #3435

### CODE REVIEW

**Comment by JonathanGiles** [comment]
Yes, there are both: https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#failOnError

Therefore, the track one libraries should be configured with this set to false, and the track two libraries will inherit the default fail on error and fail on warnings config.

**Comment by conniey** [comment]
@JonathanGiles Thanks for that tip. I've enabled the default behaviour to not fail the build. But enabled it for azure-common and applicationconfig since we own these libraries.

**Comment by JonathanGiles** [comment]
I actually want to do the opposite (as mentioned in Teams) - fail by default, opt out in track one.

**Comment by conniey** [comment]
I wanted to see which ones failed, and if any were simple enough to resolve. (ie. If there were only a few links broken.) I'll look through the failures now and change this back to WIP.

**Comment by JonathanGiles** [comment]
Awesome! Eventually we want the same pattern for spotbugs and checkstyle too.

File: keyvault/data-plane/azure-keyvault-cryptography/pom.xml [hunk 1]
```diff
@@ -1,100 +1,102 @@
-<!-- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under 
-	the MIT License. See License.txt in the project root for license information. -->
+<!-- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under
```
Comment (conniey) [inline keyvault/data-plane/azure-keyvault-cryptography/pom.xml]: Only thing changed in this file is indentation. GitHub isn't doing this diff well.

**Comment by conniey** [comment]
@JonathanGiles How does this look now? I've fixed the errors and it should build now.

**Comment by JonathanGiles** [comment]
This has bitrotted since being opened. I propose filing an issue and closing this PR. I want to have a big cleanup in pom.client.xml around JavaDoc, so this can be part of that.

**Pull request closed without merging**